common/Darwin/DarwinThreads.cpp

```diff
@@ -94,6 +94,57 @@ u64 Threading::GetThreadCpuTime()
 	return us;
 }
 
+Threading::ThreadHandle::ThreadHandle() = default;
+
+Threading::ThreadHandle::ThreadHandle(const ThreadHandle& handle)
+	: m_native_handle(handle.m_native_handle)
+{
+}
+
+Threading::ThreadHandle::ThreadHandle(ThreadHandle&& handle)
+	: m_native_handle(handle.m_native_handle)
+{
+	handle.m_native_handle = nullptr;
+}
+
+Threading::ThreadHandle::~ThreadHandle() = default;
+
+Threading::ThreadHandle Threading::ThreadHandle::GetForCallingThread()
+{
+	ThreadHandle ret;
+	ret.m_native_handle = pthread_self();
+	return ret;
+}
+
+Threading::ThreadHandle& Threading::ThreadHandle::operator=(ThreadHandle&& handle)
+{
+	m_native_handle = handle.m_native_handle;
+	handle.m_native_handle = nullptr;
+	return *this;
+}
+
+Threading::ThreadHandle& Threading::ThreadHandle::operator=(const ThreadHandle& handle)
+{
+	m_native_handle = handle.m_native_handle;
+	return *this;
+}
+
+u64 Threading::ThreadHandle::GetCPUTime() const
+{
+	return getthreadtime(pthread_mach_thread_np((pthread_t)m_native_handle));
+}
+
+void Threading::ThreadHandle::SetThreadName(const char* name) const
+{
+	// Doesn't appear to be possible to set another thread's name in OSX...
+}
```

Comment:
Is there any reason to use this over `SetNameOfCurrentThread` which works on more platforms?

Reply:
Originally my intent was to switch pxThread over to these wrappers, and I wrote the other implementations.. before I realized macos couldn't do it. Probably just easier to drop this one.